test/integration/controller/jobset_controller_test.go

```diff
@@ -638,20 +637,22 @@ func matchJobsNodeSelectors(js *jobset.JobSet, nodeSelectors map[string]map[stri
 	// Count number of updated jobs
 	jobsUpdated := 0
 	for _, job := range jobList.Items {
-		nodeSelectorKey, ok := job.Labels[jobset.ReplicatedJobNameKey]
+		rjobName, ok := job.Labels[jobset.ReplicatedJobNameKey]
 		if !ok {
 			return false, fmt.Errorf(fmt.Sprintf("%s job missing ReplicatedJobName label", job.Name))
 		}
-		if !apiequality.Semantic.DeepEqual(job.Spec.Template.Spec.NodeSelector, nodeSelectors[nodeSelectorKey]) {
+		key := nodeSelectorKey(js.Name, rjobName)
```

Comment:
instead of this, just change the key to use the replicatedJob name: https://github.com/kubernetes-sigs/jobset/blob/11db48caf22438a98807091a4fb1cc711b7d6805/test/integration/controller/jobset_controller_test.go#L182

you also need to change: https://github.com/kubernetes-sigs/jobset/blob/11db48caf22438a98807091a4fb1cc711b7d6805/test/integration/controller/jobset_controller_test.go#L609 to use only the replicatedjob name

Reply:
done